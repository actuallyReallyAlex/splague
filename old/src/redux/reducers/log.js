// Log Reducer

const logReducerInitialState = []

const logReducerDevelopmentState = []

let logState

if (process.env.REACT_APP_ENVIRONMENT === 'dev') {
  logState = logReducerDevelopmentState
} else {
  logState = logReducerInitialState
}

export default (state = logState, action) => {
  switch (action.type) {
    case 'ADD_LOG_ITEM':
      const { logItem } = action.payload
      return [...state, logItem]
    default:
      return state
  }
}
