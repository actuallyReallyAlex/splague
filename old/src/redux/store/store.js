import { createStore, combineReducers } from 'redux'
import cureReducer from '../reducers/cure'
import logReducer from '../reducers/log'
import plagueReducer from '../reducers/plague'
import playerReducer from '../reducers/player'
import uiReducer from '../reducers/ui'
import worldReducer from '../reducers/world'

const store = createStore(
  combineReducers({
    cure: cureReducer,
    log: logReducer,
    plague: plagueReducer,
    player: playerReducer,
    ui: uiReducer,
    world: worldReducer
  }),
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
)

export default store
