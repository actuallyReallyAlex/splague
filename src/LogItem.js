import React from 'react'
import { Comment } from 'semantic-ui-react'
import PropTypes from 'prop-types'

const authorAndTextStyle = {
  color: 'rgba(255, 255, 255, 0.87)'
}

const metaDataStyle = {
  color: 'rgba(255, 255, 255, 0.4)'
}

const LogItem = props => {
  return (
    <Comment className="animated fadeInUp" style={props.style}>
      <Comment.Avatar src="http://i.pravatar.cc/300" />
      <Comment.Content>
        <Comment.Author style={authorAndTextStyle}>{props.author}</Comment.Author>
        <Comment.Metadata>
          <div style={metaDataStyle}>{props.metaData}</div>
        </Comment.Metadata>
        <Comment.Text style={authorAndTextStyle}>{props.text}</Comment.Text>
      </Comment.Content>
    </Comment>
  )
}

LogItem.propTypes = {
  style: PropTypes.object,
  author: PropTypes.string,
  metaData: PropTypes.string,
  text: PropTypes.string
}

export default LogItem
