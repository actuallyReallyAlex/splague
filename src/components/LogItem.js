import React from 'react'
import PropTypes from 'prop-types'
import { Box, Text } from 'grommet'
import { User } from 'grommet-icons'

export const LogItem = ({ additionalInfo, description, icon, time, title }) => {
  const Icon = icon
  return (
    <Box flex={false} className="animated fadeInUp" direction="row" gap="small" pad="small">
      <Box
        align="center"
        height="40px"
        justify="center"
        round="small"
        style={{ background: 'rgb(40, 41, 62)' }}
        width="40px"
      >
        <Icon color="white" />
      </Box>
      <Box width="small">
        <Box justify="between" direction="row" fill="horizontal">
          <Text size="small" weight="bold">
            {title}
          </Text>
          <Text size="small">{time}</Text>
        </Box>

        {additionalInfo && (
          <Text color="grey" margin="none" size="small">
            {additionalInfo}
          </Text>
        )}
        <Text size="small">{description}</Text>
      </Box>
    </Box>
  )
}

LogItem.propTypes = {
  additionalInfo: PropTypes.string,
  description: PropTypes.string.isRequired,
  icon: PropTypes.func,
  time: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}

LogItem.defaultProps = {
  icon: User
}

export default LogItem
