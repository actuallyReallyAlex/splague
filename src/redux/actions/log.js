export const ADD_LOG_ITEM = 'ADD_LOG_ITEM'

/**
 * Adds a log item to the Log.
 * @param {Object} logItem LogItem details.
 */
export const addLogItem = logItem => {
  return {
    type: ADD_LOG_ITEM,
    payload: {
      logItem
    }
  }
}
