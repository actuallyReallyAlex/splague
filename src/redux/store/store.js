import { createStore, combineReducers } from 'redux'
import WorldReducer from '../reducers/world'

const store = createStore(
  combineReducers({
    world: WorldReducer
  }),
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
)

export default store
