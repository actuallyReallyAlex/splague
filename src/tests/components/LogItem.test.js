import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Provider } from 'react-redux'
import store from '../../redux/store/store'
import LogItem from '../../components/LogItem'
import log from '../fixtures/log'

describe('<LogItem />', () => {
  it('Should render snapshot of LogItem component.', () => {
    const { additionalInfo, description, icon, title } = log[0]
    const component = renderer
      .create(
        <Provider store={store}>
          <LogItem
            additionalInfo={additionalInfo}
            description={description}
            icon={icon}
            title={title}
          />
        </Provider>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
