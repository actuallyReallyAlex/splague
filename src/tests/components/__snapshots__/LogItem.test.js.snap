// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogItem /> Should render snapshot of LogItem component. 1`] = `
.c2 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: white;
  stroke: white;
}

.c2 g {
  fill: inherit;
  stroke: inherit;
}

.c2 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c2 *[stroke*="#"],
.c2 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c2 *[fill-rule],
.c2 *[FILL-RULE],
.c2 *[fill*="#"],
.c2 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 12px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  height: 40px;
  width: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 12px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  width: 192px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0px;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 12px;
}

.c6 {
  font-size: 14px;
  line-height: 20px;
  font-weight: bold;
}

.c7 {
  font-size: 14px;
  line-height: 20px;
}

.c8 {
  margin: 0px;
  font-size: 14px;
  line-height: 20px;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  color: grey;
}

@media only screen and (max-width:768px) {
  .c0 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    border-radius: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding: 0px;
  }
}

<div
  className="animated fadeInUp c0"
>
  <div
    className="c1"
    style={
      Object {
        "background": "rgb(40, 41, 62)",
      }
    }
  >
    <svg
      aria-label="Alert"
      className="c2"
      viewBox="0 0 24 24"
    >
      <path
        d="M12,17 L12,19 M12,10 L12,16 M12,3 L2,22 L22,22 L12,3 Z"
        fill="none"
        stroke="#000"
        strokeWidth="2"
      />
    </svg>
  </div>
  <div
    className="c3"
  />
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <span
        className="c6"
        size="small"
      >
        Test Item 1
      </span>
      <span
        className="c7"
        size="small"
      />
    </div>
    <span
      className="c8"
      size="small"
    >
      Additional Info
    </span>
    <span
      className="c7"
      size="small"
    >
      Test Description
    </span>
  </div>
</div>
`;
